### DISCUSSION TITLE
Orientation issues

### DISCUSSION THREAD

**Discussion by AbdulAlaqil** [Ergo-Tools/ActiPASS#57] [Help and Q&A (processing, quality checking or statistics)] [answered]
Dear ActiPASS Community,

I hope you are all well.

I have been processing my accelerometry data in ActiPASS (using ProPASS mode) and have encountered some participant data issues. I would like to know if anyone has faced a similar issue or knows how to handle it without losing the data.

The problem is that I have noticed some data showing an unusually high "SleepInterval" (>21 hours) in ActiPASS, while using PAL analysis has yielded different results (please see an example below). 
<img width="684" alt="Screenshot 2023-10-24 172714" src="https://github.com/Ergo-Tools/ActiPASS/assets/148870938/5c43f986-ad29-45eb-a672-0b7078431147">
![thumbnail_ScR0538 - 2023 - Week 30](https://github.com/Ergo-Tools/ActiPASS/assets/148870938/be249535-240c-4e39-8d40-1ed08608478a)

Could someone please provide an explanation?

Thank you very much,

Abdul

**Comment by pasan-h** [comment]
> Dear ActiPASS Community,
> 
> I hope you are all well.
> 
> I have been processing my accelerometry data in ActiPASS (using ProPASS mode) and have encountered some participant data issues. I would like to know if anyone has faced a similar issue or knows how to handle it without losing the data.
> 
> The problem is that I have noticed some data showing an unusually high "SleepInterval" (>21 hours) in ActiPASS, while using PAL analysis has yielded different results (please see an example below).

Hi Abdul, could you please follow the instructions below and post the correct quality check figure and batch-settings file. It's easier to understand what's going on with those extra information. 
https://github.com/Ergo-Tools/ActiPASS/discussions/55

**Reply by AbdulAlaqil** [reply]
Hi Pasan, 

Please see below the correct quality check figure, and batch setting file.

![ActiPASS batch settings](https://github.com/Ergo-Tools/ActiPASS/assets/148870938/81348e01-5451-4b77-a9e0-fcebad251cbc)

Quality check
![ScR0538-QC-1](https://github.com/Ergo-Tools/ActiPASS/assets/148870938/79885e0b-65c5-4c44-9d76-58b08cb26e53)

**Reply by pasan-h** [reply, marked as answer]
Thanks 👍. It seems ActiPASS cannot identify device orientations properly (auto flips/rotations detection has failed). This means whole activity detection is wrong. So this is **not just a sleep-interval problem**. Also I can see that you have specified both "Flipped" and "Rotated" as true. Why is that?
Could you rerun this file again with "Flipped" and "Rtated" checkboxes unticked and see what happens? For activpal devices we should not normally need to tick "Rotated" and "Flipped" checkboxes.
Also see: https://github.com/Ergo-Tools/ActiPASS/wiki/ActiPASS-FAQ-and-troubleshooting#if-actipass-has-automatic-orientation-detection-why-is-it-necessary-to-specify-orientation

**Reply by AbdulAlaqil** [reply]
Thanks, Pasan. I believe it works now correctly and I can see more activites. Just to clarify. for ActivPAL devices, do you recommend to uncheck the "Flipped" and "Rated" checkboxes?
![ScR0538-QC-1](https://github.com/Ergo-Tools/ActiPASS/assets/148870938/b4954275-e1b1-4d2a-ab2c-5b7445f43bfa)

**Reply by pasan-h** [reply]
Yes, for ActivPAL devices both boxes should be unchecked. Perhaps a long time ago, we had to mark "Flipped" checkbox for ActivPAL devices. But not anymore. If we know that devices are worn in other orientations than the defaults, then we can use those checkboxes. For the default orientations see:
https://github.com/Ergo-Tools/ActiPASS/wiki/ActiPASS-accelerometer-default-orientations

**Comment by camillebiemans-oss** [comment]
Hello everyone,

I'm encountering a similar problem as AbdulAlaqil.

For some of my files, it keeps giving the quality check error 'NoWalking'. When looking at the data in PALanalysis, there did occur walking during the day for these people. However, once processed in ActiPASS, I get the error 'NoWalking'. 
PALanalysis screenshot:
<img width="1906" height="1005" alt="PALwalking" src="https://github.com/user-attachments/assets/174e2814-60e8-4dec-b2ac-c51e5c8f5b25" />

I also got the orientation warning: 'Selected default orientations for the batch... Rotated (upside-down) = 100%'.
Once I've changed the orientation towards rotated (screenshot), I still get the 'NoWalking' error and the visual inspections showed no difference. Underneath one example (step count = 0):

<img width="1793" height="903" alt="Nowalking" src="https://github.com/user-attachments/assets/527240bd-c02b-465e-8b32-7de0dd40590c" />

Batch Settings with orientation on 'rotated':
[ActiPASS_BatchSettings.txt](https://github.com/user-attachments/files/22551224/ActiPASS_BatchSettings.txt)

Batch Settings with orientation on 'default mode' (so unthicked):
[ActiPASS_BatchSettings.txt](https://github.com/user-attachments/files/22551230/ActiPASS_BatchSettings.txt)

Could someone please think along with us? That would be great. Thank you in advance!

Camille

**Reply by pasan-h** [reply]
Hi,
For PAL devices (if you follow manufacture recommended orientation), run ActiPASS with both rotated and flipped check-boxes unticked. First process all files with this default settings. Then find out which files in the batch has wrong-classifications (no-walk etc). Try to process only those files with rotated checkbox clicked (in a separate batch, but use the same project folder).

This behaviour is expected (for a few cases) when some devices are attached up side down. If you cannot fix them (if the classification doesn't make sense) just exclude them from the analysis. Similarly even with PAL-analysis one can expect some misclassifications when users wear devices in awkward orientations (or loose devices), but perhaps they are not always apparent like in ActiPASS.
BR
Pasan

**Reply by pasan-h** [reply]
> Could someone please think along with us? That would be great. Thank you in advance!

Also try enabling advanced mode and then disabling auto orientation corrections (under advanced settings) for these problematic cases.

**Comment by camillebiemans-oss** [comment]
Dear @pasan-h,

Thank you so much for your reply. I'm trying to run the advanced mode, and use the files that you've sended me. I've followed all steps for [activating the advanced license](https://github.com/Ergo-Tools/ActiPASS/wiki/Renew-an-existing-license). But unfortunately, the 'advanced options' do not appear, see below:
<img width="1158" height="749" alt="image" src="https://github.com/user-attachments/assets/21ae60b0-f795-4490-a0d9-8cf042f31b68" />

The files have been updated, and I've restarted the ActiPASS software:
<img width="826" height="278" alt="image" src="https://github.com/user-attachments/assets/5b5f01be-3efd-40ff-8184-eba7f34d9b2a" />

The only difference that I can find with my data and [ the explanation](https://github.com/Ergo-Tools/ActiPASS/wiki/Renew-an-existing-license) is that I don't have the file 'ActiPASSQC.conf'. Could that be the problem?

Thank you so much again, really appreciated.

Camille

**Comment by pasan-h** [comment]
Hi Camille,
Did you disable ProPASS mode? Advanced mode is enabled once you disable ProPASS mode.
See FAQ:
https://github.com/Ergo-Tools/ActiPASS/wiki/ActiPASS-FAQ-and-troubleshooting#how-do-i-enable-actipass-advanced-mode-of-processing

Pasan

**Comment by camillebiemans-oss** [comment]
Thank you so much!!! This works. @pasan-h

**Comment by pasan-h** [comment]
@camillebiemans-oss 
I've had time to think about the issue you encountered (differences between ActiPASS and PAL analysis). The problem could be also caused by the cohort-profile. If participants do not have a normal walking gait (like healthy adults) but somewhat weak walking posture (elderly or sick), ActiPASS fails to detect walking and thus it's auto reference-position corrections go wrong. This can also lead to the situation seen above.

In this case I think ActiPASS is not capable correctly distinguishing behaviours. Then we can disable some features and use default reference positions like seen below. Otherwise just stick to PAL Analysis 😀.

<img width="469" height="336" alt="image" src="https://github.com/user-attachments/assets/6f2bdd2b-814f-4afd-a1cb-3f432a4cb93b" />

**Comment by Suesmith12** [comment]
Hi @pasan-h, Thanks for your helpful comments re the above.

We have one participant in our sample (N=475) whose QC checks are showing a lot of errors. Using the PAL software, this participant has 7 valid days of wear. However, when processed with ActiPASS, only one valid day of wear is detected. This seems to be related to not detecting any walking/movement.

<img width="892" height="679" alt="Screenshot 2026-06-10 150434" src="https://github.com/user-attachments/assets/0b631e27-ddef-44ca-b1a3-510c02e7b8bd" />

<img width="1224" height="689" alt="Screenshot 2026-06-10 143643" src="https://github.com/user-attachments/assets/e82821be-daa4-4d0f-801b-5c13dd267de0" />

[ActiPASS_BatchSettings.txt](https://github.com/user-attachments/files/28797350/ActiPASS_BatchSettings.txt)

I've tried your above suggestions and have found that it might be that it was worn upside down (screenshot of ActiPASS rotation warning), but even when I check this box in the settings and then process the data, the attached QC visualisation has no movement in it still and it still is saying that there is only 1 valid day when I go to the stats generation.
<img width="1219" height="697" alt="Screenshot 2026-06-10 144302" src="https://github.com/user-attachments/assets/f6a0866c-79f0-4c23-a56e-6e96c1ef5dc4" />

<img width="721" height="387" alt="Screenshot 2026-06-10 144841" src="https://github.com/user-attachments/assets/0b5f128f-6221-4a5f-abf6-53b08b3662e7" />

<img width="1190" height="695" alt="Screenshot 2026-06-10 150159" src="https://github.com/user-attachments/assets/cd15bf6f-67b8-466e-9ca0-533fe77c3ce5" />


I have also tried your suggestion above by changing the settings in case it's a cohort profile reason, but then got an error when I tried to generate any statistics for this (see attached).
<img width="711" height="389" alt="Screenshot 2026-06-10 142846" src="https://github.com/user-attachments/assets/bfb95819-98ed-4ead-b7d0-48a06e3572bd" />

Do you have any suggestions on how to proceed please? Ideally we don't want to remove this participant from our analysis, but we need at least 3 full days of wear to be included.

**Comment by pasan-h** [comment]
Hi @Suesmith12, It's frustrating when two software behave differently, isn't it? Anyway, I don't have a short answer. I assume the file is a converted CSV file by PALAnalysis or PALBatch. Otherwise (if it's an old ActivPAL version 3 datax file), then please convert to CSV and see what's happen.  

 Perhaps we might be able to what's happening by debugging while ActiPASS is processing the actual problematic file. I am on vacation until end of July, perhaps we can give it a try if the actual raw data file can be shared after striping identifiable data.
Best,
P.H.

**Reply by Suesmith12** [reply]
Hi Pasan, Thanks for getting back to me. Yes, we are using the CSV file from PALBatch. That would be great. I can follow up at the end of July - enjoy!
Best wishes,
Sue
